### ISSUE TITLE
Abnormal file size of with variable length variables

### ISSUE THREAD

**Issue by valschmidt** [Unidata/netcdf4-python#547]
I've no idea if this is actually a bug, but it is so odd that I thought I'd comment. 

The following code segment produces a file with just two variable length variables, "timestamp" and "data". The former holds a double, the latter holds a variable length blob of binary data (in this case just 100 bytes). 

```python
from netCDF4 import Dataset
import numpy
import time
f = Dataset('text.nc','w')
timestamp_dim = f.createDimension('timestamp_dim',None)
data_dim = f.createDimension('data_dim',None)
data_t = f.createVLType('u1','variable_data_t')
timestamp = f.createVariable('timestamp','d','timestamp_dim')
data = f.createVariable('data',data_t,'data_dim')
buf = numpy.ones((1,100),'uint8')
timestamp[0] = time.time()
data[0] = buf
f.close()
```
The resulting file is a whopping 73 MB!!!.

Modify the two createDimension() lines to specify a variable of length of 1, and the resulting file is just 6 kbytes. 

Is this expected??? Is there some error in my usage?

**Comment by jswhit** [comment]
I don't get 73 MB here, more like 80K:

    [mac28:/Volumes/Drobo/python] jwhitaker% ls -l text.nc
    -rw-r--r--  1 jwhitaker  climate  80016 Apr  5 13:57 text.nc

What are you hdf5 and netcdf library versions?

**Comment by valschmidt** [comment]
I'm glad to hear you don't have the same problem. 

These are from the latest Anaconda distribution:
```
libnetcdf                 4.3.3.1                       3  
netcdf4                   1.2.2               np110py27_0  
hdf5                      1.8.15.1                      2 
```

I found that the problem is caused by the default chunksizes. If I don't set it explicitly they look like this:
```
ncdump -hs text.nc
netcdf text {
types:
  ubyte(*) variable_data_t ;
dimensions:
	timestamp_dim = UNLIMITED ; // (1000 currently)
	data_dim = UNLIMITED ; // (1000 currently)
variables:
	double timestamp(timestamp_dim) ;
		timestamp:_Storage = "chunked" ;
		timestamp:_ChunkSizes = 524288 ;
	variable_data_t data(data_dim) ;
		data:_Storage = "chunked" ;
		data:_ChunkSizes = 4194304 ;
		data:_NoFill = "true" ;

// global attributes:
		:_Format = "netCDF-4" ;
}
```

But if I set them to something like 1024, the resulting file size is much more appropriate.

**Comment by jswhit** [comment]
The default chunksizes in netcdf 4.4 must have changed:

    [mac28:/Volumes/Drobo/python] jwhitaker% ncdump -hs text.nc
    netcdf text {
    types:
      ubyte(*) variable_data_t ;
    dimensions:
	    timestamp_dim = UNLIMITED ; // (1 currently)
	    data_dim = UNLIMITED ; // (1 currently)
    variables:
	    double timestamp(timestamp_dim) ;
	    	    timestamp:_Storage = "chunked" ;
		    timestamp:_ChunkSizes = 512 ;
		    timestamp:_Endianness = "little" ;
	    variable_data_t data(data_dim) ;
		    data:_Storage = "chunked" ;
		    data:_ChunkSizes = 4096 ;
		    data:_NoFill = "true" ;

    // global attributes:
		    :_Format = "netCDF-4" ;
    }

   [mac28:/Volumes/Drobo/python] jwhitaker% python checkversion.py
    netcdf4-python version: 1.2.3
    HDF5 lib version:       1.8.16
    netcdf lib version:     4.4.0-development
    numpy version           1.11.0b2

**Comment by DennisHeimbigner** [comment]
I asked Russ Rew about this. Yes, they changed from 4megabyte to 4kilobyte.
=Dennis Heimbigner
   Unidata

**Comment by jswhit** [comment]
OK, closing then...

**Comment by WardF** [comment]
A follow up from Russ Rew:
-----

Looking back a this thread, valschmidt was using library version

> libnetcdf 4.3.3.1

so the bug encountered was fixed in the 4.4.0-rc3 and subsequent releases.

----
So, there's a specific release where it was changed.  Thanks Russ!
